src/coreclr/gc/gcpriv.h

```diff
@@ -5752,7 +5752,7 @@ class heap_segment
     // swept_in_plan_p can be folded into gen_num.
     bool            swept_in_plan_p;
     int             plan_gen_num;
-    int             survived;
+    size_t          survived;
     int             old_card_survived;
     int             pinned_survived;
```

Comment:
does `pinned_survived` needs to be looked at too?

Reply:
pinned_survived is only used on basic regions so it doesn't need to be size_t. but as I mentioned, we should then make sue the basic region size cannot be too large. that should be included in this PR.